include/yaml-cpp/emitter.h

```diff
@@ -177,7 +178,9 @@ inline Emitter& Emitter::WriteStreamable(T value) {
     }
   }
 
-  if (!special) {
+  if (static_cast<float>(static_cast<int>(value)) == value){
+    stream << std::fixed << std::setprecision(1) << value;
```

Comment:
I *think* I see what you're trying to do, but could you please add a test that demonstrates this change?